### DISCUSSION TITLE
How about a "-" suffix on "b" that performs "git checkout -"?

### DISCUSSION THREAD

**Discussion by jannotti** [magit/magit#5023] [Ideas]
It's obviously not a big deal, but having just learned that "git checkout -" exists, I find I'd like the same convenience under the branch transient prefix.

**Comment by kyleam** [comment]
A similar suggestion came up a long time ago in gh-1560.

`magit-read-other-branch-or-commit` defaults to the previous branch, so `b b RET` should give you the functionality you're after (unless point is on something else that overrides the default).

**Reply by jannotti** [reply]
`b b Ret` seems good enough that I can see why `b -` isn't going to happen.  For me, it doesn't work because I set up my status buffer to show branches and find that's where I spend most of my "cursor time", so I'm almost always parked on a branch, making _that_ the default. I guess I can get used to hitting `home b b Ret`.

**Reply by kyleam** [reply]
Fwiw here's how you could add this suffix on your end.  (I'm avoiding `-` because has a hidden `-r` option, but you could use that for the key if you don't care about `-r`.)

```elisp
(defun my/magit-checkout-previous-branch (&optional args)
  "Check out the previous branch."
  (interactive (list (magit-branch-arguments)))
  (magit-run-git "checkout" args "-"))

(transient-append-suffix 'magit-branch "l"
  '("L" "Check out previous" my/magit-checkout-previous-branch))
```

**Reply by jannotti** [reply]
Thanks!
